calico_node/startup/startup.go

```diff
@@ -308,17 +325,14 @@ func waitForConnection(ctx context.Context, c client.Interface) {
 	log.Info("Datastore connection verified")
 }
 
-// writeStartupEnv writes out the startup.env file to set environment variables
-// that are required by confd/bird etc. but may not have been passed into the
-// container.
-func writeStartupEnv(nodeName string) {
-	text := "export NODENAME=" + nodeName + "\n"
-
-	// Write out the startup.env file to ensure required environments are
-	// set (which they might not otherwise be).
-	if err := ioutil.WriteFile("startup.env", []byte(text), 0666); err != nil {
-		log.WithError(err).Info("Unable to write to startup.env")
-		log.Warn("Unable to write to local filesystem")
+// writeNodeConfig writes out the this node's configuration to disk for use by other components.
+// Specifically, it creates:
+// - /var/lib/calico/nodename - used to persist the determined node name to disk for future use.
+func writeNodeConfig(nodeName string) {
```

Comment:
Just wondering if we could/should have kept the env file, seems like a nice approach since we could add more variables in there later if needed.  I guess it makes it a little trickier to read the file, it would need ot be sources before starting the binary?

Reply:
We could also add more variables to the `/var/lib/calico` directory if needed - it felt wrong to be writing both files with the same information, and it also felt wrong to start mounting the startup.env directory to the host.

This seemed like a nicer approach to me, but LMK if you think it would be better to mount startup.env to `/var/lib/calico/startup.env` or something.

Reply:
Let's stick with the nodename file.  No point making this over-general when we've only got one variable